spring-integration-core/src/test/java/org/springframework/integration/config/xml/BarrierParserTests-context.xml

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:int="http://www.springframework.org/schema/integration"
+	xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
+		http://www.springframework.org/schema/integration http://www.springframework.org/schema/integration/spring-integration.xsd">
+
+	<int:channel id="in">
+		<int:queue />
+	</int:channel>
+
+	<int:barrier id="barrier1" input-channel="in" output-channel="out" correlation-strategy-expression="'foo'" 
+				timeout="10000">
+		<int:poller fixed-delay="100" />
+	</int:barrier>
+
+	<int:channel id="out">
+		<int:queue />
+	</int:channel>
+
+	<int:channel id="release" />
+
+	<int:outbound-channel-adapter channel="release" ref="barrier1.handler" method="release" />
```

Comment:
I feel I'm fine with the solution.
Even if some one doesn't like to use `<outbound-channel-adapter>` on the matter, he can `inject` that `MessageHandler` to invoke the `release()` directly.
Although we must think about extraction some interface on the matter to cover the `Proxy` case.